src/app/tasks/task-details/task-details.component.ts

```diff
@@ -1,28 +1,111 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
-import { Task, emptyTask } from '../task.model';
+import { Component, OnInit } from '@angular/core';
+import { Task, Option, emptyTask } from '../task.model';
 import { ActivatedRoute, Router } from '@angular/router';
 import { TaskService } from '../task.service';
+import { FormControl, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { KeyValuePipe } from '@angular/common';
 
 @Component({
   selector: 'app-task-details',
   standalone: true,
-  imports: [],
+  imports: [
+    ReactiveFormsModule,
+    KeyValuePipe
+  ],
   templateUrl: './task-details.component.html',
   styleUrl: './task-details.component.scss'
 })
 export class TaskDetailsComponent implements OnInit {
 
-  task: Task | null = null;
+  task: Task = emptyTask;
+  editMode: boolean = false;
+  taskTypes : Option[] = this.taskService.getTaskTypes();
+  taskStatuses : Option[] = this.taskService.getTaskStatuses();
+
+  editTaskForm: FormGroup = new FormGroup({
+    title: new FormControl(),
+    description: new FormControl(),
+    type: new FormControl(),
+    status: new FormControl()
+  });
 
   onDeleted(taskId: number) {
     this.taskService.deleteTask(taskId);
     this.router.navigate(['tasks']);
   }
 
+  onEdit() {
+    this.editMode = true;
+  }
+  
+  onSubmit () {
+    // Get filled out form data using form group
+
+    if(this.task.id) {
+      this.task = this.taskService.updateTask(
+        this.task.id,
+        {
+          id: this.task.id,
+          title: this.editTaskForm.get('title')?.value,
+          description: this.editTaskForm.get('description')?.value,
+          type: this.taskTypes.find( taskType => 
+            taskType.value == this.editTaskForm.get('type')?.value
+          ) ?? null,
+          createdOn: new Date(),
+          status: this.taskStatuses.find(taskStatus => 
+            taskStatus.value == this.editTaskForm.get('status')?.value
+          ) ?? null,
+        }
+      );
+    }
+
+    this.router.navigate(['tasks']);
+  }
+
   constructor(private router: Router, private route: ActivatedRoute, private taskService: TaskService) {}
 
   ngOnInit(): void {
-    this.task = this.taskService.getTask(this.route.snapshot.params['id']);
+    
+    this.task = this.taskService.getTask(this.route.snapshot.params['id']) ?? emptyTask;
```

Comment:
to have 100% certainty here, you can subscribe to the activatedROute params, get id param and then get the task, that way you won't have to give it a default value of emptyTask